static-site/.eslintrc.yaml

```diff
@@ -1,2 +1,28 @@
-# This empty ESLint configuration file disables Gatsby's default ESLint rules.
-# ESLint is configured at the top level of this repository.
+# Prevent parent configuration from being applied.
+root: true
+
+extends:
+  - plugin:react/recommended
+  - plugin:react-hooks/recommended
+
+settings:
+  react:
+    version: "16"
+
+ignorePatterns:
+  - public/
+
+rules:
+  react/jsx-key: off
+  react/jsx-no-target-blank: off
+  react/no-unescaped-entities: off
+  react/no-unknown-property: off
+  react/prop-types: off
+  react-hooks/exhaustive-deps: off
```

Comment:
I'd suggest enabling this rule -- missing deps is such as common problem with hooks

Reply:
This is mostly done. There's one which I haven't dug deep enough to know how to best address: e17c5dca9b4d69699b1a92e2408d4f70f05b5295

@joverlee521 you might have a better sense of this part of the code.

Reply:
Hmm, I think it's fine to disable the rule for hooks that are only expected to be run once. 

(Also, this should be raising a similar warning for `getOverview` within `edit-overview-form.jsx`)

Reply:
The expectation of running once is what this is rule addresses - in theory it should run every time props are updated, not just initial load (see https://github.com/facebook/react/issues/14920#issuecomment-470945051).

In practice, I ran this locally and couldn't find any cases where the props would change after initial load. If there was a button to go directly from editing one group's settings to another, maybe this would start to break, but currently it seems fine from a bit of testing.

I'll add a comment to disable with this reasoning.

Reply:
> Also, this should be raising a similar warning for `getOverview` within `edit-overview-form.jsx`

That's odd, I see the warning locally but not in CI logs. Will look into it.

Reply:
From reading the code the idiomatic way to do this is to memoise the function. I don't think you can use `useMemo` for an async function, but you can achieve the same effect with `useEffect`. In this case it's simpler (and arguably clearer) to disable the linting & add a comment 👍

Reply:
Comments added in 1c111308942e2793c40172b8bf9d61ecfc7786bb.